pkg/api/api.go

```diff
@@ -10,13 +10,34 @@ type ErrorResp struct {
 // BlockVolumeCreateReq represents gluster block
 // volume create request
 type BlockVolumeCreateReq struct {
-	HaCount            int      `json:"ha"`
-	AuthEnabled        bool     `json:"auth-enabled"`
-	FullPrealloc       bool     `json:"full-prealloc"`
-	Size               uint64   `json:"size"`
-	Storage            string   `json:"storage"`
-	RingBufferSizeInMB uint64   `json:"ring-buffer-size-mb"`
-	Hosts              []string `json:"hosts"`
+	// Name represents block Volume name
+	Name               string      `json:"name,omitempty"`
+	HostingVolume      string      `json:"hosting_volume,omitempty"`
+	HaCount            int         `json:"ha,omitempty"`
+	AuthEnabled        bool        `json:"auth-enabled"`
+	FullPrealloc       bool        `json:"full-prealloc"`
+	Size               interface{} `json:"size,omitempty"`
+	Storage            string      `json:"storage,omitempty"`
+	RingBufferSizeInMB uint64      `json:"ring-buffer-size-mb,omitempty"`
+	Hosts              []string    `json:"hosts,omitempty"`
+}
+
+// BlockVolumeCreateResp represents  gluster block
+// volume create resp
+// Note: same as gluster-block cli output
+type BlockVolumeCreateResp struct {
+	IQN     string   `json:"IQN"`
+	Portals []string `json:"PORTAL(S)"`
+	Result  string   `json:"RESULT"`
+}
+
+// BlockVolumeCreateError respresents error in creating a
+// gluster block volume
+// Note: same as gluster-block cli output
+type BlockVolumeCreateError struct {
```

Comment:
this can be the generic error structure, I think this will be useful for other operations also